### PR TITLE
[CALCITE-4585]when run RelNode error,Confuse log info be thrown(xiong duan)

### PR DESCRIPTION
**Pull request by NobiGo**
This prepareStatement method capture the exception,But Only the output of SQL exception is processed.When run RelNode 
 error, The print log is "Error while preparing statement [null]", There is no useful information in this log So need to add the procedure to handle extra Relnode exception.

### CODE REVIEW

**Review by hsyuan — APPROVED** [review]
+1

File: core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java [hunk 1]
```diff
@@ -141,4 +167,29 @@ private void runQuery(String sql) throws SQLException {
...
+      final RelNode relNode = builder
+          .scan("test", "entries")
+          .project(builder.call(SqlStdOperatorTable.ABS, builder.field("name")))
+          .build();
+      runQuery(relNode);
+      fail("Query badEntries should result in an exception");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage(),
+          equalTo("java.sql.SQLException: Error while preparing statement [\n"
+              + "LogicalProject($f0=[ABS($1)])\n"
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java:190]: Hmm, the plan may be helpful but it is not a statement ?

Comment (NobiGo) [reply core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java:190]: Statement in Calcite may be  include three types: Sql. RelNode .  Queryable.  This method Called prepareStatement_. My be we listen to others opinion.

Comment (NobiGo) [reply core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java:190]: Statement in Calcite may be  include three types: Sql. RelNode .  Queryable.  This method Called prepareStatement_. My be we listen to others opinion.

File: core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java [hunk 2]
```diff
@@ -141,4 +167,29 @@ private void runQuery(String sql) throws SQLException {
...
+
+  @Test void testValidRelNodeQuery() throws SQLException {
+    final RelNode relNode = builder
+        .scan("test", "entries")
+        .project(builder.field("name"))
+        .build();
+    runQuery(relNode);
+  }
+
+  @Test void testRelNodeQueryException() throws SQLException {
```
Comment (zabetak) [inline core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java:179]: There are test failures in CI (Windows). Please fix the problem with the line endings; check other tests cases comparing plans.

**Review by zabetak — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java (hunk 2 above)
Comment (NobiGo) [reply core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java:179]: Yes. I will fix this.

File: core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java (hunk 1 above)
Comment (Aaaaaaron) [reply core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java]: Actually, I'm +1 for Danny, maybe change "statement" to "plan" is better. But the code before is also "Error while preparing statement"

Comment (NobiGo) [reply core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java]: Ok. I will change this to plan. Thanks for all of your advice.

**Comment by julianhyde** [comment]
We don't generally use `System.lineSeparator()`; we generally use "\n" in strings. To make the tests pass in Windows, use `Matchers.isLinux` in your test.

The commit message (and JIRA subject) is not clear. Can you improve it?

I don't think it will be clear to future maintainers what is the purpose of the two test methods.  Please add javadoc comments to those methods indicating what you are testing.

**Comment by NobiGo** [comment]
Hi, @julianhyde  I have try to modify this. Thanks for your advice. This really helped me a lot

**Comment by julianhyde** [comment]
@NobiGo, I refactored a little in https://github.com/julianhyde/calcite/tree/4585-relRunner-exception, and also fixed related bug https://issues.apache.org/jira/browse/CALCITE-4591. Let me know if there are any objections. I will squash & merge shortly.

**Comment by NobiGo** [comment]
@julianhyde  [CALCITE-4591](https://issues.apache.org/jira/browse/CALCITE-4591) makes this issue more complete. I'm +1 for this

**Pull request closed without merging**